common/oauthTokenManager.go

```diff
@@ -244,42 +248,64 @@ func (uotm *UserOAuthTokenManager) UserLogin(tenantID, activeDirectoryEndpoint s
 	if activeDirectoryEndpoint == "" {
 		activeDirectoryEndpoint = DefaultActiveDirectoryEndpoint
 	}
+	var dc *azidentity.DeviceCodeCredential
+	var err error
+	var record azidentity.AuthenticationRecord
+	var token azcore.AccessToken
 
-	// Init OAuth config
-	oauthConfig, err := adal.NewOAuthConfig(activeDirectoryEndpoint, tenantID)
-	if err != nil {
-		return err
+	// The conditional statement checks whether 'persist' is true before configuring token caching options.
+	// If 'persist' evaluates to true, the options for token caching are provided.
+	if persist {
+		record, err = retrieveRecord()
+		if err != nil {
+			return err
+		}
+		dc, err = azidentity.NewDeviceCodeCredential(&azidentity.DeviceCodeCredentialOptions{
+			AuthenticationRecord: record,
+			TenantID:             tenantID,
+			TokenCachePersistenceOptions: &azidentity.TokenCachePersistenceOptions{
+				AllowUnencryptedStorage: true,
+				Name:                    TokenCache,
+			}})
+		// If record is empty
+		if record == (azidentity.AuthenticationRecord{}) {
```

Comment:
'record' is created at line 2589, then we try to get a token. Post that we check whether record is empty or not. We shall check record is empty or not first as in case of empty record. NewDeviceCodeCredential is just taking it as input to fetch token. If we call that before the record is populate how is it working?
- fetch record
- create create wth record (which is empty in the first place)
- if record is empty, populate the record
- so in this flow DeviceCredential is alredy created 'dc' with empty reocrd. How does populating the record now make any difference to 'dc'.

Reply:
resolved in the latest commit.